### PR TITLE
Feature/content-loader

### PR DESCRIPTION
**Pull request by petrKavulok**
https://user-images.githubusercontent.com/79644454/151851159-5ebd5476-ff02-4f6d-8b18-8ba0d51f5b7b.mov

### CODE REVIEW

**Comment by petrKavulok** [comment]
https://user-images.githubusercontent.com/79644454/154093111-75b2593f-897c-4305-b4d1-aeb1cad3a67c.mp4

@Pe5h4 , @titkov49  what do you think? docs coming soon

File: src/containers/ContentLoader.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import ContentLoader from "react-content-loader";
+
+export function ChartLoader({title = 'Loading Data', backgroundColor = "#f3f3f3", 
+	foregroundColor = "#ecebeb", speed = 1.2, width = "100%", height = 150 }, props) {
```
Comment (titkov49) [inline src/containers/ContentLoader.js:5]: Can you please write it on different lines please? I guess that three parameters per line will look good

File: src/containers/ContentLoader.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+							<rect x={20 * i} y={rand} rx="5" ry="5" width="10" height={height-rand} key={i}/> 
+							)
+					})
+				}
+			</ContentLoader>
+	);
+}
+
+export function CellContentLoader({rows = 10, cols = 4, title = 'Loading Data',
+	backgroundColor = "#E4E5E6", foregroundColor = "#ffffff", speed = 1.2, header = true, size = 'lg' }, props) {
```
Comment (titkov49) [inline src/containers/ContentLoader.js:33]: Split those lines as well please

File: src/containers/ContentLoader.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+					})
+				} 			
+				{
+					rectangles.map((rectangle, i) => {
+						return (
+							<rect x={rectangle[0]} y={rectangle[1]} rx="3" ry="3" width={rectangle[2]} height={rectangle[3]} key={i} />
+						)
+					})
+				}  
+			</ContentLoader>
```
Comment (titkov49) [inline src/containers/ContentLoader.js:135]: As I see you used here too many tabs for this whole `return` block

File: src/containers/ContentLoader.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+	return (
+			<ContentLoader
+				title={title}
+				speed={speed}
+				viewBox={viewBox}
+				backgroundColor={backgroundColor}
+				foregroundColor={foregroundColor}
+				animate={true}
+				{...props}
+				>
```
Comment (titkov49) [inline src/containers/ContentLoader.js:120]: Closing tag should be aligned with openning one. Not with props

File: src/containers/ContentLoader.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+				{header &&
+					headRect.map((rectangle, i) => {
+						return (
+							<rect x={rectangle[0]} y={rectangle[1]} rx="3" ry="3" width={rectangle[2]} height={rectangle[3]} key={i} />
```
Comment (titkov49) [inline src/containers/ContentLoader.js:124]: Split this line into more (one prop per line)

File: src/containers/ContentLoader.js [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+						)
+					})
+				} 			
+				{
+					rectangles.map((rectangle, i) => {
+						return (
+							<rect x={rectangle[0]} y={rectangle[1]} rx="3" ry="3" width={rectangle[2]} height={rectangle[3]} key={i} />
```
Comment (titkov49) [inline src/containers/ContentLoader.js:131]: Split this line into more (one prop per line)

File: src/containers/ContentLoader.js [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+				{
+					rectangleQty.map((el, i) =>  {
+						let rand = Math.floor(Math.random() * (height * 0.7))
+						return (
+							<rect x={20 * i} y={rand} rx="5" ry="5" width="10" height={height-rand} key={i}/> 
+							)
+					})
+				}
+			</ContentLoader>
+	);
```
Comment (titkov49) [inline src/containers/ContentLoader.js:29]: As I see you used here too many tabs for this whole return block

File: src/containers/ContentLoader.js [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+				viewBox={`0 0 ${width} ${height}`}
+				backgroundColor={backgroundColor}
+				foregroundColor={foregroundColor}
+				{...props}
+			>
+				{
+					rectangleQty.map((el, i) =>  {
+						let rand = Math.floor(Math.random() * (height * 0.7))
+						return (
+							<rect x={20 * i} y={rand} rx="5" ry="5" width="10" height={height-rand} key={i}/>
```
Comment (titkov49) [inline src/containers/ContentLoader.js:24]: Split this line into more (one prop per line)

File: src/containers/ContentLoader.js [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+	}
+	let viewBox = `0, 0, ${width}, ${y + ((rows ) * 60)}`;
+
+
+	for (let i = 0; i < rows ; i++){
+		if(cols === 1) {
+			rectangles.push([width*0.05 , y + 20 + (i * 55), width*0.9 , 15],
+							)
+			headRect.push([0, 12, width * 0.15, 33],
+						[width*0.85, 12, width * 0.15, 33],
```
Comment (titkov49) [inline src/containers/ContentLoader.js:52]: As I see you used here too many tabs for those arguments. Also you should break line before first argument as well:
```js
headRect.push(
  [0, 12, width * 0.15, 33],
  [width*0.85, 12, width * 0.15, 33],
)
```

File: src/containers/ContentLoader.js [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+	);
+}
+
+export function CellContentLoader({rows = 10, cols = 4, title = 'Loading Data',
+	backgroundColor = "#E4E5E6", foregroundColor = "#ffffff", speed = 1.2, header = true, size = 'lg' }, props) {
+
+	let rectangles = [],
+		width = size === 'lg' ? 900 : 400,
+		headRect = [[0, 0, width, 17],
+					[0, 35, width, 17]],
```
Comment (titkov49) [inline src/containers/ContentLoader.js:38]: As I see you used here too many tabs for those arguments. Also you should break line after first `[` as well:
```js
headRect = [
  [0, 0, width, 17],
  [0, 35, width, 17]
]
```

File: src/containers/ContentLoader.js (hunk 9 above)
Comment (titkov49) [reply src/containers/ContentLoader.js:52]: And fix that below as well

**Review by titkov49 — CHANGES_REQUESTED** [review]

File: src/containers/ContentLoader.js [hunk 11]
```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import ContentLoader from "react-content-loader";
+
+export function ChartLoader({
+	title = 'Loading Data', backgroundColor = "#f3f3f3",
```
Comment (Pe5h4) [inline src/containers/ContentLoader.js:5]: @petrKavulok title should use i18n that it can be translated

File: src/containers/ContentLoader.js [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+				)
+		})}
+		</ContentLoader>
+	);
+}
+
+export function CellContentLoader({
+	rows = 10, cols = 4, title = 'Loading Data', 
+	backgroundColor = "#E4E5E6", foregroundColor = "#ffffff", speed = 1.2, 
+	header = true, size = 'lg' }, props) {
```
Comment (Pe5h4) [inline src/containers/ContentLoader.js:44]: @petrKavulok header should be false by default. I am not even sure, if we need that (if I understand from your video, its the first line in the loader, which has inverted colors?)

File: src/containers/ContentLoader.js [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+						key={i}
+					/> 
+				)
+		})}
+		</ContentLoader>
+	);
+}
+
+export function CellContentLoader({
+	rows = 10, cols = 4, title = 'Loading Data',
```
Comment (Pe5h4) [inline src/containers/ContentLoader.js:42]: @petrKavulok Title should use i18n

File: src/containers/ContentLoader.js [hunk 14]
```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import ContentLoader from "react-content-loader";
+
+export function ChartLoader({
```
Comment (Pe5h4) [inline src/containers/ContentLoader.js:4]: @petrKavulok How this will look on small / medium / large cards and screens?

File: demo/package.json [hunk 15]
```diff
@@ -24,7 +24,7 @@
 		"fs-extra": "^9.0.0",
 		"html-webpack-plugin": "^4.5.2",
 		"interpolate-html-plugin": "^4.0.0",
-		"node-sass": "^4.14.1",
+		"sass": "^1.44.0",
```
Comment (Pe5h4) [inline demo/package.json:27]: @petrKavulok You want to add `react-content-loader` to the packages.json of demo app

**Review by Pe5h4 — CHANGES_REQUESTED** [review]

**Review by titkov49 — APPROVED** [review]
Approved from my side

File: src/containers/ContentLoader.js (hunk 14 above)
Comment (petrKavulok) [reply src/containers/ContentLoader.js:4]: https://user-images.githubusercontent.com/79644454/154438699-31b38524-c32d-4b50-858a-08de35bc7c32.mov

File: src/containers/ContentLoader.js (hunk 12 above)
Comment (petrKavulok) [reply src/containers/ContentLoader.js:44]: yes, that's what the first line with inverted colors was supposed to demonstate

**Review by Pe5h4 — APPROVED** [review]
@petrKavulok Looks nice, dont forget for the documentation

**Review by ateska — APPROVED** [review]

**Comment by petrKavulok** [comment]
@Pe5h4 @titkov49 your thoughts on the Docs, gents?..

File: doc/content-loader.md [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+- parameters
+	- title (string / optional) - accepts already translated version
+	- backgroundColor (string / optional) - color of loader's cells, in HEX format (#f3f3f3)
+	- foregroundColor (string / optional) - color of flare, in HEX format (#ffffff)
+	- speed (number / optional) - animation speed in seconds
+	- width (number or string(percentage) / optional) - sets max width of the viewbox - default 100%, should work best in most cases
+	- height (number / optional) - sets height of the viewbox
+	- others - all other props accepted by react-content-loader (https://www.npmjs.com/package/react-content-loader#options)
+- example:
+```
```
Comment (titkov49) [inline doc/content-loader.md:48]: Same with ` ```js `

File: doc/content-loader.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
+#  ASAB WebUI ContentLoader Component
+
+- Customisable content loader component to be used before actual data is rendered
+- requires 'react-content-loader' to be installed 
+
+## CellContentLoader
+
+- parameters
```
Comment (Pe5h4) [inline doc/content-loader.md:8]: @petrKavulok Can you please specify defaults of those parameters?

File: doc/content-loader.md [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+	)
+} 
+```
+
+
+## ChartLoader
+
+- Substitues actual data in bar charts
+- Chart is created randomly upon every page refresh
+- parameters
```
Comment (Pe5h4) [inline doc/content-loader.md:39]: @petrKavulok Defaults here again, please

File: src/containers/ContentLoader.js [hunk 19]
```diff
@@ -0,0 +1,174 @@
...
+					/> 
+				)
+		})}
+		</ContentLoader>
+	);
+}
+
+export function CellContentLoader({
+	rows = 1, cols = 3, title = '', 
+	backgroundColor = "#E4E5E6", foregroundColor = "#ffffff", speed = 1.2,
```
Comment (Pe5h4) [inline src/containers/ContentLoader.js:43]: @petrKavulok Default speed might be better `2`, `1.2` seems to me too fast

File: src/containers/ContentLoader.js [hunk 20]
```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import ContentLoader from "react-content-loader";
+
+export function ChartLoader({
+	title = '', backgroundColor = "#f3f3f3", 
+	foregroundColor = "#fff", speed = 1.2, width = "100%",
```
Comment (Pe5h4) [inline src/containers/ContentLoader.js:6]: @petrKavulok Default speed might be better `2`, `1.2` seems to me too fast

**Review by Pe5h4 — CHANGES_REQUESTED** [review]

**Review by Pe5h4 — APPROVED** [review]

**Pull request merged**